src/components/layouts/footer/index.tsx

```diff
@@ -35,56 +36,64 @@ const Footer = () => {
 
   // handle submit
   const handleSubmit = async () => {
-    setLoading(true);
-
     if (email === "") {
+      setError(true);
+
       toast({
         title: "Error",
         description: toastDesc,
         variant: "destructive",
       });
       setLoading(false);
+
       return;
     }
+    setLoading(true);
 
-    // const payload = {
-    //   email: values,
-    // };
-
-    try {
-      const apiUrl = await getApiUrl();
-      const response = await axios.post(
-        `${apiUrl}/api/v1/newsletter-subscription`,
+    const apiUrl = await getApiUrl();
+    await axios
+      .post(
```

Comment:
This was not the agreed endpoint

Reply:
That is the endpoint in the swagger docs 
![Screenshot_20240816-090031_Chrome](https://github.com/user-attachments/assets/18bbc9dd-946f-4ec6-b2f4-de46a27b075e)